module2/service/src/main/java/com/epam/esm/service/validator/Validator.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.esm.service.validator;
+
+import com.epam.esm.service.dto.GiftCertificateDto;
+import com.epam.esm.service.dto.TagDto;
+import org.springframework.stereotype.Component;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class Validator {
```

Comment:
It's better to split validator into 2 classes. Now this class violates single responsibility principle

Reply:
fixed